openassessment/fileupload/views.py

```diff
@@ -0,0 +1,151 @@
+import hashlib
+import json
+import os
+
+from django.conf import settings
+from django.shortcuts import HttpResponse, Http404
+from django.utils import timezone
+from django.views.decorators.http import require_http_methods
+
+from . import exceptions
+from .backends.filesystem import is_upload_url_available, is_download_url_available
+from .backends.base import Settings
+
+
+@require_http_methods(["PUT", "GET"])
+def filesystem_storage(request, key):
+    """
+    Uploading and download files to the local filesystem backend.
+    """
+    if isinstance(key, unicode):
+        key = key.encode("utf-8")
+    if request.method == "PUT":
+        if not is_upload_url_available(key):
+            raise Http404()
+        content, metadata = get_content_metadata(request)
+        save_to_file(key, content, metadata)
```

Comment:
`save_to_file` can throw an exception. Perhaps you want to catch it and return something other than a 500 error?